### ISSUE TITLE
Problem when some folder names contain a - (minus)

### ISSUE THREAD

**Issue by bknaepen** [pytest-dev/pytest#8173]
If I run pytest within the test folder with the following structure I have no problems:

```
tests/
  __init__.py
  Pipe/
    __init__.py
    test_exec.py
  PipeCheb/
    __init__.py
    test_exec.py
  PipePoiseuille/
    __init__.py
    test_exec.py
```

However, if some of the folders have a `-` in their names, Iike this,

 ```
tests/
  __init__.py
  Pipe/
    __init__.py
    test_exec.py
  Pipe-Cheb/
    __init__.py
    test_exec.py
  Pipe-Poiseuille/
    __init__.py
    test_exec.py
```

I then get the following error:

```
============================= test session starts ==============================
platform darwin -- Python 3.8.5, pytest-6.1.1, py-1.9.0, pluggy-0.13.1
rootdir: /Users/bknaepen/Desktop/test
collected 2 items / 1 error / 1 selected                                       

==================================== ERRORS ====================================
________________ ERROR collecting Plane-Poiseuille/test_exec.py ________________
import file mismatch:
imported module 'test_exec' has this __file__ attribute:
  /Users/bknaepen/Desktop/test/Pipe-Cheb/test_exec.py
which is not the same as the test file we want to collect:
  /Users/bknaepen/Desktop/test/Plane-Poiseuille/test_exec.py
HINT: remove __pycache__ / .pyc files and/or use a unique basename for your test file modules
=========================== short test summary info ============================
ERROR Plane-Poiseuille/test_exec.py
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 0.07s ===============================
```

Is this a known issue?

Cheers,
Bernard.

**Comment by bluetech** [comment]
I highly recommend not using a `-` in these directory names because a `-` is not valid in a Python identifier, and you are using Python packages (the `__init__.py` files indicate this). Nonetheless, I'd be interested to know if using `pytest --import-mode=importlib` fixes or changes the error in any way?

**Comment by symonk** [comment]
Using hyphenated package names is highly discouraged in python in general, it creates a special case for importing etc, would recommend naming those packages with an underscore at the very least 👍

**Comment by bknaepen** [comment]
@bluetech @symonk Thanks a lot for your answers, it's very kind.

Our (small) project currently had one test in several of these directories and I want to move to a more robust test procedure, hence I am looking into learning pytest. I had put the __init__.py file in these directories because all the tests have at this stage the same filename. But as you pointed out, I created this other issue because of the `-`character in the packages paths.

I ran the command `pytest --import-mode=importlib`  and it works with the `-` in the paths. It also works if I remove all the __init__.py files.

Thanks again for pointing me in the right direction,

Cheers,
Bernard.
